netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -249,13 +249,21 @@ public void operationComplete(ChannelFuture future) throws Exception {
       throw new IOException("Failed to bind", future.cause());
     }
     channel = future.channel();
-    channel.eventLoop().execute(new Runnable() {
-      @Override
-      public void run() {
-        listenSocketStats = new ListenSocket(channel);
-        channelz.addListenSocket(listenSocketStats);
-      }
-    });
+
+    // We should never throw any Exception after this point to abide by the contract of
+    // ServerListener.serverShutdown()
+
+    try {
+      channel.eventLoop().execute(new Runnable() {
+        @Override
+        public void run() {
+          listenSocketStats = new ListenSocket(channel);
+          channelz.addListenSocket(listenSocketStats);
+        }
+      });
+    } catch (RuntimeException e) {
+      log.log(Level.WARNING, "Error while starting server", e);
```

Comment:
This does not seem good. I don't think we should go in a direction that logs exceptions instead of propagating them.

Reply:
When the eventloop.execute() is broken, there is not much we can handle. What about 
```java
try {
      channel.eventLoop().execute(new Runnable() {
        @Override
        public void run() {
          listenSocketStats = new ListenSocket(channel);
          channelz.addListenSocket(listenSocketStats);
        }
      });
    } catch (RuntimeException e) {
      try {
        channel.close();
      } finally {
        channel = null;
        throw e;
      }
    }
```